geomstats/geometry/poincare_ball.py

```diff
@@ -317,3 +317,268 @@ def inner_product_matrix(self, base_point=None):
+
+        means_units = gs.expand_dims(means, 0)
+
+        means_units = gs.repeat(means_units, mesh_data_units.shape[0], axis=0)
+
+        distance_to_mean = metric(mesh_data_units, means_units)
+        variances_units = gs.expand_dims(variances, 0)
+        variances_units = gs.repeat(variances_units,
+                                    distance_to_mean.shape[0], axis=0)
+
+        distribution_normal = gs.exp(-(distance_to_mean ** 2)
+                                     / (2 * variances_units ** 2))
+
+        zeta_sigma = PI_2_3 * variances
+        zeta_sigma = zeta_sigma * gs.exp((variances ** 2 / 2)
+                                         * gs.erf(variances / gs.sqrt(2)))
+
+        result_num = gs.expand_dims(mixture_coefficients, 0)
+        result_num = gs.repeat(result_num,
+                               len(distribution_normal), axis=0)
+        result_num = result_num * distribution_normal
+        result_denum = gs.expand_dims(zeta_sigma, 0)
+        result_denum = gs.repeat(result_denum,
+                                 len(distribution_normal), axis=0)
+
+        result = result_num / result_denum
+
+        return result
+
+
+class Normalization_Factor_Storage():
+    """A class for computing the normalization factor."""
+
+    def __init__(self, variances, dimension):
+
+        self.dimension = dimension
+
+        self.variances = variances
+        self.normalisation_factor_var = \
+            self.normalization_factor(variances, dimension)
+
+        cond_1 = self.normalisation_factor_var.sum() != \
+            self.normalisation_factor_var.sum()
+        cond_2 = self.normalisation_factor_var.sum() == float('+inf')
+        cond_3 = self.normalisation_factor_var.sum() == float('-inf')
+
+        if cond_1 or cond_2 or cond_3:
+            print('WARNING :\n'
+                  'untracktable normalisation factor :')
+            max_nf = len(variances)
+
+            limit_nf = ((self.normalisation_factor_var /
+                         self.normalisation_factor_var)
+                        * 0).nonzero()[0].item()
+            self.variances = self.variances[0:limit_nf]
+            self.normalisation_factor_var = \
+                self.normalisation_factor_var[0:limit_nf]
+            if cond_1:
+                print('\t Nan value in processing normalisation factor')
+            if cond_2 or cond_3:
+                print('\t +-inf value in processing normalisation factor')
+
+            print('\t Max variance is now : ', self.variances[-1])
+            print('\t Number of possible variance is now: '
+                  + str(len(self.variances)) + '/' + str(max_nf))
+
+        _, log_grad_zeta = \
+            self.norm_factor_gradient(self.variances, dimension)
+
+        self.phi_inv_var = self.variances ** 3 * log_grad_zeta
+
+    def find_normalisation_factor(self, variance):
+        """Given a variance, finds the normalisation factor."""
+        N, P = variance.shape[0], self.variances.shape[0]
```

Comment:
no uppercase letters for variables, no single letter for variable: maybe find names that are more explicit and help understanding the code?

Reply:
Not resolved?